core/sodasql/scan/test_result.py

```diff
@@ -28,43 +28,49 @@ class TestResult:
 
     def __str__(self):
         if self.passed:
-            status_str = 'passed'
+            status_str = "passed"
         elif self.skipped:
-            status_str = 'skipped'
+            status_str = "skipped"
         else:
-            status_str = 'failed'
-        return (f'Test {self.test.title} {status_str}' +
-                (f" with group values {self.group_values}" if self.group_values else '') +
-                f' with measurements {json.dumps(JsonHelper.to_jsonnable(self.values))}')
+            status_str = "failed"
+        return (
+            f"Test {self.test.title} {status_str}"
+            + (f" with group values {self.group_values}" if self.group_values else "")
+            + f" with measurements {json.dumps(JsonHelper.to_jsonnable(self.values))}"
+        )
 
     def to_dict(self) -> dict:
         if not self.test or not self.test.expression:
-            return {
-                'error': 'Invalid test result'
-            }
+            return {"error": "Invalid test result"}
 
         test_result_json = {
-            'id': self.test.id,
-            'title': self.test.title,
-            'description': self.test.title,  # for backwards compatibility
-            'expression': self.test.expression
+            "id": self.test.id,
+            "title": self.test.title,
+            "description": self.test.title,  # for backwards compatibility
+            "expression": self.test.expression,
         }
 
         if self.test.column:
-            test_result_json['columnName'] = self.test.column
+            test_result_json["columnName"] = self.test.column
+
+        if self.test.source:
+            test_result_json["source"] = self.test.source
+        else:
+            test_result_json["source"] = "soda-sql"
```

Comment:
This is never going to get executed since we added a default value to source property. We either need to change the default value to `soda-sql` or add it to the test creation inside parser.py (L329 - L333)

Reply:
@Antoninj I added a default value, and  informed @bastienboutonnet to fix it there - https://github.com/sodadata/soda-sql/blob/main/core/sodasql/scan/test.py#L27